astropy/io/fits/tests/test_image.py

```diff
@@ -1506,7 +1506,7 @@ def test_scale_back_compressed(self):
     def test_lossless_gzip_compression(self):
         """Regression test for https://aeon.stsci.edu/ssb/trac/pyfits/ticket/198"""
 
-        noise = np.random.normal(size=(1000, 1000))
+        noise = np.random.normal(size=(20, 20))
```

Comment:
Why not go all the way to `(2, 2)`? :smirk:

Reply:
Also does it really need to be random?

Reply:
Because some floats can be represented exactly by the quantization of the compression algorithms. For 4 values it's realistically possible that all 4 are exactly representable (thus failing the test that makes sure the default loses precision), however for 800 it's extremely unlikely.

Reply:
Yeah, it would probably be better to use some fixed values. 😓

Reply:
Okay, but it seems to be non-trivial to construct a good example here without dumping a huge array hard-coded in there. That's because the quantization uses the noise of each row to scale the values to integers. So too few values will make the quantization "exact". 

Gosh, at some point I really have to look into how the quantization works exactly 😭 

Will leave this as-is for now. The main point was generating much less values to make it faster without making the test flaky.

Reply:
Okay, fine by me. Thanks for looking into this! :smile:

Reply:
But just to be on the safe (reproducible) side: I use a seed so the random values are predictable.

Reply:
Yes I think that you need enough values to test the quantization properly. I think the current change is a good compromise.